examples/sample_lib/start_runtime_with_sample_lib.py

```diff
@@ -0,0 +1,59 @@
+# Standard
+from pathlib import Path
+import json
+import os
+import shutil
+import sys
+import tempfile
+
+# First Party
+import alog
+
+# Local
+from caikit.runtime.__main__ import main
+from caikit.runtime.dump_services import dump_grpc_services, dump_http_services
+import caikit
+
+if __name__ == "__main__":
+    with tempfile.TemporaryDirectory() as workdir:
+        caikit.config.configure(
+            config_dict={
+                "merge_strategy": "merge",
+                "runtime": {
+                    "library": "sample_lib",
+                    "local_models_dir": workdir,
+                    "lazy_load_local_models": True,
+                    "grpc": {"enabled": True},
+                    "http": {"enabled": True},
+                    "training": {"save_with_id": False, "output_dir": workdir},
+                },
+            }
+        )
+        # Make sample_lib available for import
+        sys.path.append(
+            os.path.join(Path(__file__).parent.parent.parent, "tests/fixtures"),
+        )
+
+        # dump protos
+        shutil.rmtree("protos", ignore_errors=True)
+        dump_grpc_services("protos")
+        dump_http_services("protos")
+
+        # create a sample.json file for training
+        with open(
+            os.path.join("protos", "sample.json"), "w", encoding="utf-8"
+        ) as handle:
+            handle.write(
+                json.dumps(
+                    [
+                        {"number": 1},
+                        {"number": 2},
+                    ]
+                )
+            )
+
+        alog.configure(default_level="debug")
+        main()
+
+        # remove generated protos
+        shutil.rmtree("protos", ignore_errors=True)
```

Comment:
NIT: Since you need to `ctrl-C` to kill a server, I suspect that this will not actually get run

Reply:
To add a hint here: https://stackoverflow.com/questions/1112343/how-do-i-capture-sigint-in-python

Should be pretty easy to register an interrupt handler to do this instead